### PR TITLE
Master-port #52441 

### PR DESCRIPTION
**Pull request by mchugh19**
cron return a single line

### What does this PR do?
Ports #52441 to master

Add get_entry to cron module to return a single cron entry. This allows for use in tools like saltcheck:

```
validate_cron.{{ task }}_exists:                                          
  module_and_function: cron.get_entry                                     
  args:                                                                   
    - root                            
    - {{ task }}                                                          
  assertion: assertEqual                                                  
  assertion_section: identifier                                           
  expected-return: {{ task }}
```

### CODE REVIEW

**Review by dhiltonp — CHANGES_REQUESTED** [review]
This PR needs some tests. 

It's a good candidate for unit tests - patch list_tab to return sample data, then verify that it breaks if cmd and identifier are specified, and works if identifier or cmd are specified.

Is it possible for the same identifier to occur multiple times in the crontab? What about the same command at different intervals?

**Review by myii — CHANGES_REQUESTED** [review]

**Comment by myii** [comment]
Apologies, I started that review weeks ago but I wsited for the PR route but both were closed.

**Comment by dwoz** [comment]
@mchugh19 Can you address the comments?

**Comment by mchugh19** [comment]
Added tests and @myii's changes. I think this should be good to go!

File: salt/modules/cron.py [hunk 1]
```diff
@@ -405,6 +408,36 @@ def list_tab(user):
...
+        Search for line with identifier
+
+    cmd:
+        Search for cron line with cmd
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' cron.identifier_exists root identifier=task1
```
[the remark this answers is not in the document]
Comment (mchugh19) [reply salt/modules/cron.py]: Corrected

[the remark this answers is not in the document]
Comment (mchugh19) [reply salt/modules/cron.py]: Corrected

File: salt/modules/cron.py [hunk 2]
```diff
@@ -405,6 +408,36 @@ def list_tab(user):
...
+    '''
+    cron_entries = list_tab(user).get('crons', False)
```
[the remark this answers is not in the document]
Comment (mchugh19) [reply salt/modules/cron.py]: Corrected

**Comment by mchugh19** [comment]
re-run pr-py3-centos-8

**Comment by mchugh19** [comment]
re-run pr-ubuntu1604-py3-m2crypto

**Comment by mchugh19** [comment]
re-run pr-ubuntu1604-py2-m2crypto

**Review by dhiltonp — DISMISSED** [review]

**Comment by mchugh19** [comment]
re-run pr-amazon2-py2

**Comment by DmitryKuzmenko** [comment]
There are no py2 anymore. I've updated the branch to force refreshing of the tests list.

**Review by dwoz — APPROVED** [review]

**Pull request merged**